documentation/modules/security/proc-configuring-internal-clients-to-trust-cluster-ca.adoc

```diff
@@ -0,0 +1,84 @@
+// Module included in the following assemblies:
+//
+// assembly-security.adoc
+
+[id='configuring-internal-clients-to-trust-cluster-ca-{context}']
+= Configuring internal clients to trust the cluster CA
+
+This procedure describes how to configure a Kafka client that resides inside the Kubernetes cluster — connecting to the `tls` listener on port 9093 — to trust the cluster CA certificate.
+
+The easiest way to achieve this for an internal client is to use a volume mount to access the `Secrets` containing the necessary certificates and keys.
+
+Follow the steps to configure trust certificates that are signed by the cluster CA for Java-based Kafka Producer, Consumer, and Streams APIs.
+
+Choose the steps to follow according to the certificate format of the cluster CA: PKCS #12 (`.p12`) or PEM (`.crt`).
+
+The steps describe how to mount the Cluster Secret that verifies the identity of the Kafka cluster to the client pod.
+
+.Prerequisites
+
+* The Cluster Operator must be running.
+* There needs to be a `Kafka` resource within the Kubernetes cluster.
+* You need a Kafka client application outside the Kubernetes cluster that will connect using TLS, and needs to trust the cluster CA certificate.
+* The client application must be running in the same namespace as the `Kafka` resource.
+
+.Using PKCS #12 format (.p12)
+
+. Mount the cluster Secret as a volume when defining the client pod.
++
+For example:
++
+[source,shell,subs="+quotes,attributes"]
+----
+kind: Pod
+apiVersion: {API_Version}
+metadata:
+  name: client-pod
+spec:
+  containers:
+  - name: client-name
+    image: client-name
+    volumeMounts:
```

Comment:
I think this should:
* Mount only the P12 file into an exact path which can be configured
* Mount the password into env var where it can be used for the Java configuration

WDYT @ppatierno?

Reply:
Yes I think it's a good idea

Reply:
I've added:

env:
    - name: SECRET_PASSWORD
    valueFrom:
      secretKeyRef:
        name: my-secret
        key: my-password